### DISCUSSION TITLE
How do put a background on the first line?

### DISCUSSION THREAD

**Discussion by trajano** [starship/starship#1880] [Q&A] [answered]
I know how to create a background for each segment, but how do you set the background to the end of the line?

**Comment by mickimnet** [comment]
@trajano what do you mean by "set the background to the end of the line"?

Do you want to set the background for the last module/segment or do you want to use the background of the shell?

**Reply by trajano** [reply]
I think the first one you said something like making the rest of the line have the some background color.  

![image](https://user-images.githubusercontent.com/110627/99185112-bd615080-2715-11eb-99bf-76a79f317b81.png)

As I wrote this I wonder if there's a thing that says put spaces until the end of the  screen then I just set the background color for it.

**Comment by mickimnet** [comment]
No, after the text "default" in your screenshot, a `newline` is entered in the output of the prompt, by the module `line_break`. There is no check how wide the terminal is.

You probably could do write a `custom` module which does that--see [Configuration/custom_command](https://starship.rs/config/#custom-commands)

**Reply by trajano** [reply]
Is there some sort of XTERM/VT escape sequence that could be passed to do the wideness thing?  I am thinking something like that should be possible considering some prompts I have seen from ohmyzsh put information at the end of the line.

In ANSI I think we can do the following

`\033[K` to erase to the end of the line so maybe combine that with something that sets the background color

**Reply by mickimnet** [reply]
Information in the prompt a the right side of the terminal, are a shell specific feature. In _Fish Shell_ there is a function `fish_right_prompt` for it and _Zsh_ has `RPROMPT`.

**Reply by trajano** [reply]
Almost got it
```
[custom.foo]
command = "echo a"  # shows output of command]
style="bg:blue"
disabled=false
when="true"
shell="sh"
#format = " transcending [$output]($style)"
# format = " \u001B\\[K  [$output]($style)"
format = "\u001B\\[46m\u001B\\[K\u001B\\[0m  [$output]($style)"
```
just have to get rid of that 'a' then probably do some work to make it less fugly by removing the ANSI escape sequences

Inched a bit closer

```
[custom.foo]
command = "echo a"  # shows output of command]
style="bg:yellow"
disabled=false
when="true"
shell="sh"
#format = " transcending [$output]($style)"
# format = " \u001B\\[K  [$output]($style)"
#format = "\u001B\\[46m\u001B\\[K\u001B\\[0m  [$output]($style)"
format = "[$output\u001B\\[K\u001B\\[0m]($style)"
```
(yellow looks fugly though)

**Comment by trajano** [comment, marked as answer]
Here's how I did it in the end.  Note the `style` has to be the same as the background to create the illusion.  It uses erase till the end of the line ANSI sequence.

```
[custom.foo]
command = "echo a"
style="yellow bg:yellow"
when="true"
shell="sh"
format = "[$output\u001B\\[K]($style)"
```

**Comment by jaens** [comment]
You generally want to set the background in the top-level format, so this is as simple as:
```toml
format = "$some_widget $some_other_widget [\u001B\\[0K](bg:green)"
```

This creates a green background from the end of the prompt to the end of the line. `\e[0K` is the terminal escape code for "clear until the end of the line" (quoted in TOML syntax).
